src/Sarif.Viewer.VisualStudio/FileWatcher/SarifFolderMonitor.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using EnvDTE;
+
+using EnvDTE80;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.Sarif.Viewer.Options;
+using Microsoft.Sarif.Viewer.Services;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.Sarif.Viewer.FileWatcher
+{
+    /// <summary>
+    /// Handles loading & monitoring sarif logs under solution directory .sarif folder.
+    /// </summary>
+    internal class SarifFolderMonitor : IDisposable
+    {
+        public const string SarifFolderName = ".sarif";
+        public const string SarifFileExtensionName = ".sarif";
+
+        private readonly IFileSystem fileSystem;
+        private IFileWatcher fileWatcher;
+        private bool filesLoaded;
+        private readonly ILoadSarifLogService sarifLoadLogService;
```

Comment:
>private readonly [](start = 8, length = 16)

group all readonly first

Reply:
done

---
In reply to: [602304497](https://github.com/microsoft/sarif-visualstudio-extension/pull/382#discussion_r602304497) [](ancestors = 602304497)